frontend/src/app/modules/common/dynamic-forms/services/dynamic-fields/dynamic-fields.service.ts

```diff
@@ -270,27 +283,37 @@ export class DynamicFieldsService {
 
   private getFieldOptions(field:IOPFieldSchemaWithKey) {
     const allowedValues = field._embedded?.allowedValues || field._links?.allowedValues;
+    let options;
 
     if (!allowedValues) {
       return;
     }
 
     if (Array.isArray(allowedValues)) {
-      const options = allowedValues[0]?._links?.self?.title ?
+      const optionValues = allowedValues[0]?._links?.self?.title ?
         this.formatAllowedValues(allowedValues) :
         allowedValues;
 
-      return of(options);
+      options = of(optionValues);
     } else if (allowedValues!.href) {
-      return this._httpClient
+      options = this.httpClient
         .get(allowedValues!.href!)
         .pipe(
           map((response:api.v3.Result) => response._embedded.elements),
           map(options => this.formatAllowedValues(options)),
         );
     }
 
-    return;
+    // Backend status options don't include the default 'not set' status
+    if (options && field.type === 'ProjectStatus') {
+      const defaultStatusOption = {
+        name: projectStatusI18n('not_set', this.I18n),
+      }
+
+      options = options.pipe(map(optionValues => ([defaultStatusOption, ...optionValues])))
+    }
+
+    return options;
```

Comment:
This code strikes me as misplaced as it is specific to the `ProjectStatus`. From the comment I understand that the assumption would be for the backend to provide the value. It currently does this via the 'required' schema property but it does not specify how that 'not set'/'none' value is to be displayed.

Since https://community.openproject.org/wp/37398 aims to introduce a generic '(none)' value, how about we alter the requirements to always include the '(none)' value in case the property is not required. That would mean, that for the project status, the 'not set'  option would be replaced by a '(none)' option. The benefit is that the user does not have to adjust to different 'not set' values for different properties. What is your take on this, @wielinde?

If having the '(none)' option in all the non required properties turns out to be to heavy, we could differentiate between the sheer existence of the default option which could be added here, and its representation which would IMO be the responsibility of the specific field.

Reply:
Great, I like the 'none' option for all the non required single-select 👍